packages/cardano-services/src/Asset/DbSyncAssetProvider.ts

```diff
@@ -66,8 +66,10 @@ export class DbSyncAssetProvider extends DbSyncProvider implements AssetProvider
     if (extraData?.history) await this.loadHistory(assetInfo);
     if (extraData?.nftMetadata)
       assetInfo.nftMetadata = await this.#dependencies.ntfMetadataService.getNftMetadata(assetInfo);
-    if (extraData?.tokenMetadata)
-      assetInfo.tokenMetadata = (await this.#dependencies.tokenMetadataService.getTokenMetadata([assetId]))[0];
+    if (extraData?.tokenMetadata) {
+      const tokenMetadata = await this.#dependencies.tokenMetadataService.getTokenMetadata([assetId]);
+      assetInfo.tokenMetadata = tokenMetadata && tokenMetadata.length > 0 ? tokenMetadata[0] : undefined;
```

Comment:
```suggestion
      assetInfo.tokenMetadata = tokenMetadata[0];
```
With a try/catch around this, the value can then be assigned as so